### PR TITLE
Allow filter facet collapsing

### PR DESCRIPTION
**Pull request by jmuscireum**
This adds functionality to make the filter facets collapsible.

### CODE REVIEW

File: src/main/java/sirius/web/controller/Facet.java [hunk 1]
```diff
@@ -143,6 +163,48 @@ public Facet addItems(Iterable<String> items) {
     }
 
     /**
+     * Enables or disables collapsing for this facet.
+     *
+     * @param facetCollapsingEnabled <tt>true</tt> to enable facet collapsing, <tt>false</tt> otherwise
+     * @return the facet itself for fluent method calls
+     */
+    public Facet setFacetCollapsingEnabled(boolean facetCollapsingEnabled) {
```
Comment (Tobsucht) [inline src/main/java/sirius/web/controller/Facet.java]: "withFacetCollapsingEnabled" to be consistent with other setters

File: src/main/java/sirius/web/controller/Facet.java [hunk 2]
```diff
@@ -143,6 +163,48 @@ public Facet addItems(Iterable<String> items) {
...
+        return facetCollapsingEnabled;
+    }
+
+    /**
+     * Sets the number of visible facet items.
+     *
+     * @param visibleFacetItems the number of visible facet items
+     * @return the facet itself for fluent method calls
+     */
+    public Facet setVisibleFacetItems(int visibleFacetItems) {
```
Comment (Tobsucht) [inline src/main/java/sirius/web/controller/Facet.java]: withVisibleFacetItems

File: src/main/java/sirius/web/controller/Facet.java [hunk 3]
```diff
@@ -152,6 +214,29 @@ public boolean hasItems() {
...
+        if (!facetCollapsingEnabled) {
+            return false;
+        }
+
+        // Don't hide anything when items are selected
+        if (!values.isEmpty()) {
+            return false;
+        }
+
+        // Don't hide anything when there is only one item to hide
```
Comment (Tobsucht) [inline src/main/java/sirius/web/controller/Facet.java]: this isn't really necessary?

File: src/main/java/sirius/web/controller/Facet.java [hunk 4]
```diff
@@ -25,6 +26,8 @@
     private String title;
     private List<String> values;
     private final ValueComputer<String, String> translator;
+    private boolean facetCollapsingEnabled = false;
```
Comment (andyHa) [inline src/main/java/sirius/web/controller/Facet.java]: redundant - visibleFacectItems controls this

File: src/main/java/sirius/web/controller/Facet.java [hunk 5]
```diff
@@ -25,6 +26,8 @@
...
+    private int visibleFacetItems;
```
Comment (andyHa) [inline src/main/java/sirius/web/controller/Facet.java]: maxVisibleFacetItems

File: src/main/java/sirius/web/controller/Facet.java (hunk 4 above)
Comment (jmuscireum) [reply src/main/java/sirius/web/controller/Facet.java]: It should be possible to completely disable collapsing. To what should visibleFacectItems be set in this case?

File: src/main/java/sirius/web/controller/Facet.java [hunk 6]
```diff
@@ -152,6 +214,29 @@ public boolean hasItems() {
...
+            return false;
+        }
+
+        // Don't hide anything when items are selected
+        if (!values.isEmpty()) {
+            return false;
+        }
+
+        // Don't hide anything when there is only one item to hide
+        if (items.size() - visibleFacetItems == 1) {
```
Comment (andyHa) [inline src/main/java/sirius/web/controller/Facet.java]: counterintuitive

**Pull request merged**